std/algorithm/iteration.d

```diff
@@ -4280,3 +4284,263 @@ unittest
+            {
+                return (item + prev_item) / 2.0;
+            }
+        }
+    }
+    else
+    {
+        foreach (index, item; r.enumerate)
+        {
+            if (index == j_int)
+            {
+                return item;
+            }
+        }
+    }
+
+    assert(0);
+}
+
+/**
+Function to compute the quantile of a $(D SortedRange) given $(D percentile).
+If the range is empty, this function will fail an assert in debug mode and
+return NaN in release mode. Is $(D O(1)).
+
+Params:
+    r = a random access range
+    percentile = where to calculate the quantile
+
+Returns:
+    the quantile of the range given $(D percentile)
+*/
+double quantile(alias less = "a < b", R)(SortedRange!(R, less) r, double percentile) if (
+    isRandomAccessRange!R &&
+    !isInfinite!R &&
+    is(ElementType!R : double))
+in
+{
+    assert(!r.empty);
+}
+body
+{
+    import std.math : modf, approxEqual;
+
+    if (r.length == 0)
+    {
+        return double.nan;
+    }
+    else if (r.length == 1)
+    {
+        return r.front;
+    }
+
+    double x = r.length * percentile;
+    real j;
+    real g = modf(x, j);
+    uint j_int = cast(uint) j;
+
+    if (g.approxEqual(0))
+    {
+        return (r[j_int - 1] + r[j_int]) / 2.0;
+    }
+    else
+    {
+        return r[j_int];
+    }
+}
+
+///
+@safe unittest
+{
+    import std.algorithm.sorting : sort;
+    import std.range : assumeSorted;
+
+    assert([1, 2, 3, 4].assumeSorted.quantile(0.25) == 1.5);
+    assert([1, 2, 3, 4].assumeSorted.quantile(0.5) == 2.5);
+    assert([1, 2, 3, 4].assumeSorted.quantile(0.75) == 3.5);
+
+    assert(sort([48, 31, 1, 37, 89, 83, 71]).quantile(0.5) == 48);
+}
+
+unittest
+{
+    import std.internal.test.dummyrange;
+    import std.range : assumeSorted;
+    import std.math : approxEqual;
+
+    DummyRange!(ReturnBy.Value, Length.Yes, RangeType.Forward) r;
+    assert(r.assumeSorted.quantile(0.25) == 3);
+    assert(r.assumeSorted.quantile(0.5).approxEqual(5.5));
+    assert(r.assumeSorted.quantile(0.75) == 8);
+}
+
+/**
+Convince function that wraps a call to the $(D quantile) function with 0.5 as
+the percentile parameter. For algorithmic complexity, refer to the docs of
+$(D quantile).
+
+Params:
+    r = a random access range or a forward range with a member length
+
+Returns:
+    the median of r
+*/
+auto median(alias less = "a < b", R)(SortedRange!(R, less) r)
+{
+    return quantile(r, 0.5);
+}
+
+///
+@safe unittest
+{
+    import std.algorithm.sorting : sort;
+    import std.range : assumeSorted;
+    import std.math : approxEqual;
+
+    assert(sort([48, 31, 1, 37, 89, 83, 71]).median == 48);
+
+    assert([3, 5, 12].assumeSorted.median == 5);
+    assert([3, 5, 7, 13, 21, 23, 23, 39].assumeSorted.median == 17);
+    assert([2.9, 6.2, 6.5, 20.5].assumeSorted.median.approxEqual(6.35));
+    assert([2.9, 5.8, 6.2, 6.5, 20.5].assumeSorted.median.approxEqual(6.2));
+
+    assert([12, 5, 3].assumeSorted!"a > b".median == 5);
```

Comment:
> I ask because I need two seperate tests for the debug and release versions.

I'd suggest just adding a bunch of asserts in debug mode since nobody is expected to compile as: `dmd -debug -release abc.d` ;)